contracts/base/strategies/iron-fold/IronFoldStrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: ISC
+/**
+* By using this software, you understand, acknowledge and accept that Tetu
+* and/or the underlying software are provided “as is” and “as available”
+* basis and without warranties or representations of any kind either expressed
+* or implied. Any use of this open source software released under the ISC
+* Internet Systems Consortium license is done at your own risk to the fullest
+* extent permissible pursuant to applicable law any and all liability as well
+* as all warranties, including any fitness for a particular purpose with respect
+* to Tetu and/or the underlying software and the use thereof are disclaimed.
+*/
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../StrategyBase.sol";
+import "../../../third_party/iron/IIronChef.sol";
+import "../../../third_party/iron/IRMatic.sol";
+import "../../../third_party/iron/CompleteRToken.sol";
+
+
+/// @title Abstract contract for Iron strategy implementation
+/// @author belbix
+abstract contract IronFoldStrategyBase is StrategyBase {
```

Comment:
we will have two strategies, one for LP pairs and this for lending
I guess need to reflect it in the name

Reply:
I thought Fold would be enough, no folding on LPs right?

Reply:
ah sure, my bad